e2e/support/helpers/e2e-misc-helpers.js

```diff
@@ -99,28 +99,29 @@ export function interceptPromise(method, path) {
  *   cy.visit(`/dashboard/1`);
  * });
  */
-const cypressWaitAllRecursive = (results, currentCommand, commands) => {
-  return currentCommand.then(result => {
-    results.push(result);
+const cypressWaitAllRecursive = (results, commands) => {
+  const [nextCommand, ...restCommands] = commands;
+  if (!nextCommand) {
+    return results;
+  }
 
-    const [nextCommand, ...rest] = Array.from(commands);
+  return nextCommand.then(result => {
+    results.push(result);
 
     if (nextCommand == null) {
-      return results;
+      return;
     }
 
-    return cypressWaitAllRecursive(results, nextCommand, rest);
+    return cypressWaitAllRecursive(results, restCommands);
   });
 };
 
 export const cypressWaitAll = function (commands) {
   const results = [];
 
-  return cypressWaitAllRecursive(
-    results,
-    cy.wrap(null, { log: false }),
-    commands,
-  );
+  return cy.wrap(results).then(() => {
```

Comment:
The fix is to just wrap the `results` which will be mutated by `cypressWaitAllRecursive` so we don't need to worry about what's being returned from `cypressWaitAllRecursive` at all. Because we're not _yielding_ any result inside `cypressWaitAllRecursive`, `commands` are already run at this point.